fvm/src/machine.rs

```diff
@@ -0,0 +1,77 @@
+use crate::context::OverrideContext;
+use crate::externs::FakeExterns;
+use fvm::machine::MachineContext;
+pub use fvm::machine::{DefaultMachine, Machine, Manifest, NetworkConfig};
+use fvm::state_tree::StateTree;
+use fvm_ipld_blockstore::MemoryBlockstore;
+
+pub struct KytheraMachine<M = DefaultMachine<MemoryBlockstore, FakeExterns>> {
+    inner: M,
+    pub(crate) override_context: OverrideContext,
+}
+
+impl<M> KytheraMachine<M>
+where
+    M: Machine,
+{
+    pub fn new(
+        context: MachineContext,
+        blockstore: MemoryBlockstore,
+        externs: FakeExterns,
+    ) -> anyhow::Result<KytheraMachine<DefaultMachine<MemoryBlockstore, FakeExterns>>> {
+        let machine = DefaultMachine::new(&context, blockstore, externs)?;
+        Ok(KytheraMachine {
+            inner: machine,
+            override_context: OverrideContext::default(),
```

Comment:
same as regarding `kernel.rs` can we leave a comment with a `TODO`, ideally I'd also put the `OverrideContext` structure here so it's all isolated.

Reply:
We can not really do that as we are instantiating a new kernel on every message. The instantiation is done in the [CallManager::send_resolved()](https://github.com/filecoin-project/ref-fvm/blob/master/fvm/src/call_manager/default.rs#L694-L703). In the case of an internal message (so from one actor to another) this is triggered by the send syscall of the `send()` syscall.

This send syscall actually leverages the `CallManager::with_transaction()` method to proceed to sending the message, so that events and gas usage are tracked.

This actually means that we have no hands on which parameters are sent in the [cm.send::<Self>()](https://github.com/filecoin-project/ref-fvm/blob/master/fvm/src/kernel/default.rs#L391-L396) method.

So, as we can not pass information between Kernel we need to store this `OverrideContext` in either the `CallManager` or the `Machine`. And as for the call manager we need the `#[repr(transparent)]` to be able to perform the unsafe code in `with_transaction()` we can not have another inner properties in the structure apart the `DefaultCallManager`. So `Machine` is used to store `OverrideContext`.